include/networkit/graph/GraphTools.hpp

```diff
@@ -406,6 +406,38 @@ void sortEdgesByWeight(Graph &G, bool decreasing = false);
  */
 std::vector<node> topologicalSort(const Graph &G);
 
+/**
+ * Given a  grah G, randomizes the weights in the graph. The weights are distributed in
+ * the range [0, 1] by default, unless a distribution is provided. This occurs in-place
+ * if the graph is already weighted, but creates a weighted copy before randomizing weights
+ * if it wasn't.
+ *
+ * Since this can be run in parallel the result can be non-deterministic. For deterministic
+ * results the function can be ran sequentially with the deterministic flag.
+ * @param   G           Directed input graph
+ * @param   deterministic  If the algorithm is to be ran sequentially or not
+ * @param   distr       Random distribution
+ */
+template <class Distribution = std::uniform_real_distribution<edgeweight>>
+void randomizeWeights(
+    Graph &G,
+    Distribution distr = std::uniform_real_distribution<edgeweight>{0, std::nexttoward(1.0, 2.0)},
+    bool deterministic = false) {
+    if (!G.isWeighted())
+        G = toWeighted(G);
+    if (!deterministic) {
+        G.parallelForEdges(
+            [&](node u, node v) { G.setWeight(u, v, distr(Aux::Random::getURNG())); });
+    } else {
+        G.forEdges([&](node u, node v) { G.setWeight(u, v, distr(Aux::Random::getURNG())); });
```

Comment:
I think we can make this parallel by using a generator for each thread and by setting a deterministic random seed for each edge. A simple idea (I am not proposing to use this exact code) could be:
```cpp
G.parallelForEdges([&](node u, node v) {
   std::mt19937 gen;
   gen.seed(std::hash<node>()(u) ^ std::hash<node>()(v) ^ Aux::Random::getSeed());
   G.setWeight(u, v, distr(gen));
});
```

Reply:
Are the edges I don't think we can set a generator for each thread considering the way `parallelForEdges` is implemented. Since getSeed isn't thread-dependant we can't use that, as which seed each thread gets is still non-deterministic.

So would need to be something like just ` gen.seed(std::hash<node>()(u) ^ std::hash<node>()(v));`, since URNG says it is thread-local, but it actually isn't (the `localSeedGeneration` is never used, it is just set to the global seed and `getSeed` just gets the global seed), so making any calls to NetworKit's seeds would make it non-deterministic.

Reply:
Well, we can just use `hash(u) * someLargePrime + hash(v)` or something similar.